learn/advanced/faceted_search.md

```diff
@@ -0,0 +1,210 @@
+      "Suppromo":1,
+      "TORCHSTAR":1,
+      "Vinkki":1,
+      "WEWILL":1,
+      "hykolity":2,
+      "iHomma":1,
+      "iLintek":1
+    },
+    "category":{
+      "Home & Kitchen":1,
+      "Tools & Home Improvement":22,
+      "Toys & Games":1
+    },
+    "rating":{
+      "0":7,
+      "1":1,
+      "3.6":1,
+      "3.8":1,
+      "4":2,
+      "4.1":2,
+      "4.4":1,
+      "4.5":2,
+      "4.6":1,
+      "4.7":3,
+      "5":3
+    }
+  }
+  …
+}
+```
+
+`facetDistribution` contains an object for every given facet, in this case, `brand`, `category`, and `rating`. For each of these facets, there is another object containing all the different values and the count of matching documents. Note that zero values will not be returned: if there are no `led` products under the `Electronics` category, it is not displayed.
+
+::: note
+By default, `facets` returns a maximum of 100 facet values for each faceted field. You can change this value using the `maxValuesPerFacet` property of the [`faceting` index settings](/reference/api/settings.md#faceting).
+:::
+
+### Facet stats
+
+When using the `facets` parameter, any matching documents with facets containing numeric values are displayed in a `facetStats` object. `facetStats` contains the numeric minimum (`min`) and maximum (`max`) values per facet for all documents matching the search query. This can be used to create a range slider component allowing users to select a range of values for a facet.
+
+If none of the matching documents have a numeric value for a facet, that facet is not included in the `facetStats` object.
+
+Meilisearch ignores string values like `"21"` when computing `facetStats`.
+
+The following response shows the `facetStats` when searching for `led`:
+
+```json
+{
+  …
+  "facetStats":{
+    "rating":{
+      "min":0.0,
+      "max":5.0
+    }
+  }
+  …
+}  
+```
+
+Since `rating` was the only numeric facet in our example, it is returned in the `facetStats` object.
+
+## Facet types
+
+### Single select facets
+
+Single-select or conjunctive facets use the `AND` logical operator. They only allow the selection of a single value per facet.
+
+Suppose your `ecommerce` index contains a facet for `brand` with the values `BrandA`, `BrandB`, and `BrandC`. Single select facets will allow you to view products from one brand at a time. If you want to view `BrandA` and `BrandB` products, you will have to try both searches individually. If you want to further narrow down results using the `reviews` facet, you would need to do so separately for both brands.
+
+The image below shows single-select facets in action. When the user searches for `led` and selects `Sunlite`, other brands are no longer visible.
+
+![Meilisearch demo for single-select facets searching for 'LED'](/faceted-search/facets-single-select.png)
+
+The code sample below shows the query for the search mentioned above:
+
+<CodeSamples id="faceted_search_conjunctive_facets_1" />
+
+### Multi-select facets
```

Comment:
cc @meilisearch/devrel-team I think this is something to explain in a tutorial for users who do not use instant-meilisearch

Reply:
Could you elaborate @gmourier ? DO you think the current guide is not enough?

Reply:
Sure! The current documentation guide does not explain how to do it; it explains the general concept from the end-users standpoint.

It's natively done by instant-meilisearch using the multi-search API endpoint. Still, if you use anything other than that, you will have to implement the glue instant-meilisearch does to achieve that, and adding that tutorial could be super helpful for users.

Reply:
I agree with @gmourier 



After reading that part, I don't think I understand the concept. Also, it is mixing two things. The normal disjunctive facet search, and the "cool behavior" facet search.

The title should mention that this part is about disjunctive facet search.

This part should then: 
1. introduce the normal disjunctive facet search.
2. Mention that you can improve the behavior by using the following concept (from chat GPT):

> Multi-select faceting can be designed in various ways, depending on the specific use case and the preferences of the users. One possible approach is to keep the facet count visible even after the user selects one or more facets. This means that the user can see how many results are associated with each facet, and how many of those results match the other facets that have been selected.

> For example, if the user selects "Nike" and "Adidas" as brands, the facet count for each brand would show how many results are associated with each brand. If there are 200 total results, and 100 are Nike shoes and 50 are Adidas shoes, the facet counts would show "Nike (100)" and "Adidas (50)". If the user then selects "sneakers" as a style, the facet counts for brands would update to show how many of the Nike and Adidas shoes are also sneakers. For example, it might show "Nike (80)" and "Adidas (30)", indicating that 80 of the 100 Nike shoes and 30 of the 50 Adidas shoes are sneakers.

> By keeping the facet count visible, the user can make more informed decisions about which facets to select or deselect, based on how many results each facet is associated with and how those results overlap with other selected facets. This can help the user quickly refine their search results to find the most relevant items.

Reply:
Added an example for the cool behavior

Reply:
The @meilisearch/docs-team discussed this internally and while we agree this guide does not cover the described implementation (aka "The Cool Behaviour"), we don't think it should. This page is supposed to be a general overview of faceting, teaching users fundamental faceting details. Our intended target are users who are not familiar with faceting.

The behaviour described by @bidoubiwa and @gmourier is relatively advanced and falls outside this goal. Trying to fit it in this article would result in a guide that gets confusing to new users towards the end (which was incidentally my own experience when reviewing it), and that does not cover the cool behaviour with the attention it deserves.

So, to hopefully reach a compromise that leaves everyone satisfied, we suggest:

1. Replace an extensive description of the behaviour with a very short paragraph and a code sample indicating the possible interactions between multi-index search and different types of facets;
2. Create a tutorial teaching users how to create the interface described by @bidoubiwa and @gmourier. We will coordinate with the @meilisearch/devrel-team to see whether this is a better fit for the blog or the docs.

Reply:
Thanks @guimachiavelli! I would be happy to hear more about the guide when you'll have time to discuss cc @Strift